src/beautifiers/remark.coffee

```diff
@@ -1,44 +1,62 @@
+###
+Requires https://github.com/remarkjs/remark
+###
 "use strict"
 Beautifier = require('./beautifier')
 
 module.exports = class Remark extends Beautifier
   name: "Remark"
-  link: "https://github.com/wooorm/remark"
+  link: "https://github.com/remarkjs/remark"
   options: {
     _: {
       gfm: true
       yaml: true
       commonmark: true
-      footnotes: true
-      pedantic: true
-      breaks: true
-      entities: true
-      setext: true
-      closeAtx: true
-      looseTable: true
-      spacedTable: true
-      fence: true
-      fences: true
-      bullet: true
-      listItemIndent: true
-      incrementListMarker: true
-      rule: true
-      ruleRepetition: true
-      ruleSpaces: true
-      strong: true
-      emphasis: true
-      position: true
+      search_for_configuration: true
     }
     Markdown: true
   }
 
-  beautify: (text, language, options) ->
-    return new @Promise((resolve, reject) ->
-      try
-        remark = require 'remark'
-        cleanMarkdown = remark().process(text, options).toString()
-        resolve cleanMarkdown
-      catch err
-        @error("Remark error: #{err}")
-        reject(err)
+  executables: [
+    {
+      name: "Remark CLI"
+      cmd: "remark"
+      homepage: "https://github.com/remarkjs/remark"
+      installation: "https://github.com/remarkjs/remark/tree/master/packages/remark-cli#installation"
+      version: {
+        parse: (text) ->
+          return text.match(/remark: (\d+\.\d+\.\d+)/)[1]
+        runOptions: {
+          returnStderr: true
+        }
+      }
+    }
+  ]
+
+  beautify: (text, language, options, context) ->
+    cwd = context.filePath and path.dirname context.filePath
+    remarkArgs = [
+      '--no-color'
+      '--no-ignore'
+      '--file-path'
+      context.filePath
+    ]
+    remarkOptions = {}
+
+    if (!options.search_for_configuration)
+      settings = Object.assign({}, options)
+      delete settings.search_for_configuration
+      serializedSettings = JSON.stringify(settings).slice(1, -1)
+      remarkArgs.push('--no-config', '--setting', serializedSettings)
+
+    return new Promise((resolve, reject) =>
```

Comment:
A new promise object is not really necessary here at this point, but it is helpful to integrate the new implementation of the beautifier with #1990